### ISSUE TITLE
Date parsing from log

### ISSUE THREAD

**Issue by telatin** [goodwright/nextflow.py#6]
When trying to run a script, I get

```
    return Execution(
  File "/Users/telatina/intelconda3/envs/xxx/lib/python3.8/site-packages/nextflow/execution.py", line 27, in __init__
    self.get_processes_from_log()
  File "/Users/telatina/intelconda3/envs/xxx/lib/python3.8/site-packages/nextflow/execution.py", line 184, in get_processes_from_log
    fields["started"] = datetime.strptime(
  File "/Users/telatina/intelconda3/envs/xxx/lib/python3.8/_strptime.py", line 568, in _strptime_datetime
    tt, fraction, gmtoff_fraction = _strptime(data_string, format)
  File "/Users/telatina/intelconda3/envs/xxx/lib/python3.8/_strptime.py", line 349, in _strptime
    raise ValueError("time data %r does not match format %r" %
ValueError: time data '2022-ept-03 06:51:53.491' does not match format '%Y-%b-%d %H:%M:%S.%f'
```

And I noticed the log shows the month with 4 chars:
```
Sept-03 15:16:40.435 [Task submitter] INFO  nextflow.Session - [88/b6a0b2] Submitted process > CIAO
```

that fools the regex in utils.py (which captures only 3 chars), and then messes with the date parsing in any case.

This is happening with `java "18.0.1.1" 2022-04-22` on macOS, while I got a three chars month (`Sep`) under Linux, `java "11.0.16" 2022-07-19`. I should have used Nextflow 21.10.6 and 22.04.5 in both cases.

I'll try to see if I find a fix on Monday but wanted to open to keep a note

**Comment by telatin** [comment]
Update: thanks to the nf-slack wisdom, the problem relies on the locale. I can prepare a PR fixing this later today

**Comment by samirelanduk** [comment]
Ah yes, I've been burned by locale issues with `datetime` before actually - good to know. Thanks!

**Comment by telatin** [comment]
I have a fix ready but didnt make a branch for the PR that is already open, if you want to merge/trash that first I'll make another :)

**Comment by samirelanduk** [comment]
I found the conversation and added the fix they suggested - thanks for checking that with them as I was thinking of a much more circuitous fix of manually correcting the date strings for each four letter month.

@alexharston and I are just finalising some stuff relating to installation, so can't address the conda PR just yet, but we will get to it.

Will release these fixes as 0.4.1 in the next few days hopefully.
